docs/modules/secret-operator/pages/secretclass.adoc

```diff
@@ -122,6 +122,34 @@ If the same AD domain _is_ shared between multiple Kubernetes clusters, the foll
 - The Kubernetes Nodes' names and fully qualified domain names
 - The Kubernetes Namespaces' names (only Namespaces that use Kerberos)
 
+[#ad-samaccountname]
+===== Custom `samAccountName` generation
+
+IMPORTANT: `samAccountName` customization is an experimental preview feature, and may be changed or removed at any time.
+
+By default, the `samAccountName` for created Active Directory accounts will be generated by Active Directory.
+
+These can be customized if required, such as for compliance with internal policies. For example, the following configuration will generate
+samAccountNames following the pattern "myprefix-abcd".
+
+[source,yaml]
+----
+spec:
+  backend:
+    kerberosKeytab:
+      admin:
+        activeDirectory:
+          experimentalGenerateSamAccountName:
+            prefix: myprefix- # <1>
+            totalLength: 13 # <2>
+----
+
+`kerberosKeytab.admin.activeDirectory.experimentalGenerateSamAccountName.prefix`:: A prefix that will be prepended to all generated `samAccountName` values.
+`kerberosKeytab.admin.activeDirectory.experimentalGenerateSamAccountName.totalLength`:: The desired length of `samAccountName` values, _including_ `prefix`. Must not be larger than 20.
```

Comment:
```suggestion
`kerberosKeytab.admin.activeDirectory.experimentalGenerateSamAccountName.totalLength`:: The desired length of `samAccountName` values, _including_ `prefix`. Must not be larger than 20. Random alphanumeric characters will be used for padding after the prefix.
```

Reply:
Padding implies an unimportant constant, IMO. Also feels like the wrong place to explain the core function of the option.

Reply:
b6aadf5d773c1dcf08acf16f0eb3fb51f358443f